### ISSUE TITLE
Declaration History Limited to 4 (ages 0-3)

### ISSUE THREAD

**Issue by kreynoldsf5** [F5Networks/f5-appsvcs-extension#6]
##### ISSUE TYPE

 - Bug Report or Documentation Update

##### AS3 BUILD/ VERSION

```
f5-appsvcs-3.0.0-34.noarch.rpm
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.0.6
  Build       0.0.3
  Edition     Point Release 6
  Date        Fri Apr 20 18:04:26 PDT 2018
```
NOTE: this is a VE with 1 core and 2GB RAM (perhaps declaration history is keyed to resources?)

##### SUMMARY

[Documentation](http://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/3/refguide/as3-api.html#api-methods-summary) states a declaration history of 16 declarations (0-15) is kept. AS3 only keeps a history of 4 declarations.

##### STEPS TO REPRODUCE

POST 4 declarations and note the output of a GET to https://{{bigip}}/mgmt/shared/appsvcs/declare?age=list
```
[
    {
        "name": "____appsvcs_declaration-1525273534246",
        "timestamp": 1525273534246,
        "date": "2018-05-02T15:05:34.246Z",
        "age": 0
    },
    {
        "name": "____appsvcs_declaration-1525273530784",
        "timestamp": 1525273530784,
        "date": "2018-05-02T15:05:30.784Z",
        "age": 1
    },
    {
        "name": "____appsvcs_declaration-1525273527654",
        "timestamp": 1525273527654,
        "date": "2018-05-02T15:05:27.654Z",
        "age": 2
    },
    {
        "name": "____appsvcs_declaration-1525273520317",
        "timestamp": 1525273520317,
        "date": "2018-05-02T15:05:20.317Z",
        "age": 3
    }
]
```
Now POST a 5th declaration and note the output of https://{{bigip}}/mgmt/shared/appsvcs/declare?age=list.
```
[
    {
        "name": "____appsvcs_declaration-1525273591802",
        "timestamp": 1525273591802,
        "date": "2018-05-02T15:06:31.802Z",
        "age": 0
    },
    {
        "name": "____appsvcs_declaration-1525273534246",
        "timestamp": 1525273534246,
        "date": "2018-05-02T15:05:34.246Z",
        "age": 1
    },
    {
        "name": "____appsvcs_declaration-1525273530784",
        "timestamp": 1525273530784,
        "date": "2018-05-02T15:05:30.784Z",
        "age": 2
    },
    {
        "name": "____appsvcs_declaration-1525273527654",
        "timestamp": 1525273527654,
        "date": "2018-05-02T15:05:27.654Z",
        "age": 3
    }
]
```
Checking the output of https://{{bigip}}/mgmt/shared/appsvcs/declare shows the 5th declaration is present -- its just not displayed in the history. 

##### EXPECTED RESULTS

Declaration history of 16 entries with ages from 0-15 to be present.
##### ACTUAL RESULTS

Declaration history was limited to 4 entries (0-3).

**Comment by kaustriaf5** [comment]
Hi @kreynoldsf5, the affected portion of the documentation will be updated. The default declaration history limit is 4, and the maximum is 15. This is in the [AS3 request schema](http://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/3/refguide/schema-reference.html#as3). To change this setting, you must include `historyLimit` everytime you `POST` the declaration, like the example below. (If you don't include it in subsequent requests, the default value of 4 will be used). 
```
{
"class":"AS3",
"action": "deploy",
"historyLimit": 15,
"declaration":  { 
  "class": "ADC",
  "schemaVersion": "3.0.0",
  "id": "declarationID",
  "controls": {
      "logLevel": "debug"
  },
  "myTenant": {
...
```

**Comment by JosephPJordan** [comment]
Thanks Kevin, we've published Document Revision 1.1 with this change (http://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/3/refguide/revision-history.html).  Closing this issue.
